sky/provision/kubernetes/instance.py

```diff
@@ -668,7 +698,14 @@ def _create_pods(region: str, cluster_name_on_cloud: str,
         # on most base images. E.g., do not use Python, since that may not
         # be installed by default.
         _check_user_privilege(namespace, context, uninitialized_pods_list)
-        _setup_ssh_in_pods(namespace, context, uninitialized_pods_list)
+        if not provider_config.get('disable_ssh', False):
+            _setup_ssh_in_pods(namespace, context, uninitialized_pods_list)
```

Comment:
Just curious, if the parallelism is taking most of the time. Can we just move the ssh setup script in to our setup commands for kubernetes:
https://github.com/skypilot-org/skypilot/blob/0f0008d41ecf4919b169541fb86a4ec60d1485ab/sky/templates/kubernetes-ray.yml.j2#L413-L431

The parallelism will only happen once then.

Reply:
Ah yeah that's a good point - we can refactor SSH setup to happen in `setup_commands`, and that way we can also waiting for SSH ready. That'll help reduce some latency.

However, most of the latency is not from the overheads of running more parallel threads, but rather the actual steps themselves (apt update, apt install openssh-server and then the service restart).

That said, I agree we should do that refactor - do you think we can do it in a separate PR? I can file an issue for it.